File: zxyhttp/src/main/java/com/zxy/http/entity/BaseBean.java
package com.zxy.http.entity;

/**
 * Created by zxy on 2021/2/27 0027 18:41
 * ******************************************
 * * 所有基类的父类
 * ******************************************
 */
public class BaseBean<T> {
    private int errorCode;
    private String errorMsg;
    private T data;

    public int getErrorCode() {
        return errorCode;
    }

    public void setErrorCode(int errorCode) {
        this.errorCode = errorCode;
    }

    public String getErrorMsg() {
        return errorMsg;
    }

    public void setErrorMsg(String errorMsg) {
        this.errorMsg = errorMsg;
    }

    public T getData() {
        return data;
    }

    public void setData(T data) {
        this.data = data;
    }
}

File: zxyhttp/src/main/java/com/zxy/http/entity/BaseBean.kt
package com.zxy.http.entity

import android.os.Parcelable
import kotlinx.android.parcel.Parcelize

/**
 * Created by zxy on 2019/9/10-11:11
 * Class functions
 * ******************************************
 * * 通用基类
 * ******************************************
 */
@Parcelize
data class BaseBean(
        var code: String? = null,
        var message: String? = null,
        var success: Boolean? = null,
        var token: String? = null,
        var userId: String? = null,
        var captchaUrl: String? = null,//图片
        var count: Int? = 0,
        var batch:String="",//批次号
        var flag: Boolean? = null,//标记
        var perCodes: MutableList<String>? = null,//权限集合
        var data: DataBean? = null//注册极光推送响应值
) : Parcelable {
    @Parcelize
    data class DataBean(
            var code: String? = null,
            var success: String? = null
    ) : Parcelable
}